numpy/_core/src/multiarray/arraytypes.c.src

```diff
@@ -366,28 +385,23 @@ NPY_NO_EXPORT int
     }
     else {
         temp = (@type@)@func2@(op);
-    }
-    if (PyErr_Occurred()) {
-        PyObject *type, *value, *traceback;
-        PyErr_Fetch(&type, &value, &traceback);
-        if (PySequence_NoString_Check(op)) {
-            PyErr_SetString(PyExc_ValueError,
-                    "setting an array element with a sequence.");
-            npy_PyErr_ChainExceptionsCause(type, value, traceback);
-        }
-        else {
-            PyErr_Restore(type, value, traceback);
+        if (temp == (@type@)-1 && PyErr_Occurred()) {
```

Comment:
This just moves it.  The first branch clearly can't fail, and the second one now has an `== -1` check (which required fixing the called helpers a bit).